src/aten/sycl/Indexing.cpp

```diff
@@ -7,17 +7,160 @@
 #include <ATen/ATen.h>
 #include <ATen/AccumulateType.h>
 #include <ATen/Dispatch.h>
-#include <ATen/native/CanUse32BitIndexMath.h>
+#include <ATen/MemoryOverlap.h>
+#include <ATen/native/Resize.h>
 #include <ATen/native/TensorIterator.h>
+#include <aten/sycl/Indexing.h>
 #include <aten/sycl/Loops.h>
-#include <aten/sycl/TensorInfo.h>
-#include <comm/SYCLContext.h>
 
-namespace at {
-namespace native {
-namespace xpu {
+#include <comm/SYCLContext.h>
+#include <comm/TensorInfo.h>
 
 using namespace at::xpu::detail;
+using namespace at::xpu;
+
+namespace at::native::xpu {
+
+template <typename ValType>
+class IndexSelectScalarFunctor {
+ public:
+  void operator()(
+      ValType* dst,
+      ValType* src,
+      int64_t dst_off,
+      int64_t src_off,
+      int64_t idx,
+      ValType alpha) const {
```

Comment:
Why are the `idx` and `alpha` not used?

Reply:
IndexKernel is an IPEX-designed kernel, since before stock CUDA implementation is not good enough. The general kernel template is shared with `aten::index_select`, `aten::index_add`, `aten::index_copy`, and `aten::index_fill` so far in IPEX. The scalar functor is abstracted to following the broadest semantics. So here, for `aten::index_select`, `idx` and `alpha` are not necessary.